substrate/frame/revive/src/tests/sol/host.rs

```diff
@@ -0,0 +1,537 @@
+
+			let Contract { addr, .. } =
+				builder::bare_instantiate(Code::Upload(code)).build_and_unwrap_contract();
+
+            let result = builder::bare_call(addr)
+                .data(
+                    HostTransientMemory::HostTransientMemoryCalls::transientMemoryTest(HostTransientMemory::transientMemoryTestCall { slot: slot, a: value })
+                        .abi_encode(),
+                )
+                .build_and_unwrap_result();
+            assert!(
+                result.flags != ReturnFlags::REVERT,
+                "test reverted"
+            );
+            assert_eq!(
+                U256::from_be_bytes::<32>(result.data.try_into().unwrap()),
+                U256::from(0),
+                "transient storage should return zero for {:?}", fixture_type
+            );
+        });
+    }
+}
+
+#[test]
+#[ignore]
+fn log_works() {
+	for fixture_type in [FixtureType::Solc, FixtureType::Resolc] {
+		let (_code, _) = compile_module_with_type("Host", fixture_type).unwrap();
+        todo!("implement this test");
+    }
+}
+
+#[test]
+#[ignore]
+fn selfdestruct_works() {
+    {
+        use pallet_revive_fixtures::HostSelfDestructEvm;
+        let fixture_type = FixtureType::Solc;
+		let (code, _) = compile_module_with_type("HostSelfDestructEvm", fixture_type).unwrap();
+
+        let expected_bobs_balance = 100_000_000_000u64;
+		
+        ExtBuilder::default().build().execute_with(|| {
+            
+            <Test as Config>::Currency::set_balance(&ALICE, 100_000_000_000);
+            <Test as Config>::Currency::set_balance(&BOB, 0);
+
+			let Contract { addr, .. } =
+				builder::bare_instantiate(Code::Upload(code)).build_and_unwrap_contract();
+            assert!(test_utils::get_contract_checked(&addr).is_some());
+            {
+                let account_id32 = <Test as Config>::AddressMapper::to_account_id(&addr);
+
+                <Test as Config>::Currency::set_balance(&account_id32, expected_bobs_balance);
+            }
+
+            {
+                let result = builder::bare_call(addr)
+                    .gas_limit(1_000_000_000.into())
+                    .data(
+                        HostSelfDestructEvm::HostSelfDestructEvmCalls::selfdestructOp(HostSelfDestructEvm::selfdestructOpCall { recipient: BOB_ADDR.0.into() })
+                            .abi_encode(),
+                    )
+                    .build_and_unwrap_result();
+                assert!(
+                    result.flags != ReturnFlags::REVERT,
+                    "test reverted"
+                );
+                
+                let bobs_balance = <Test as Config>::Currency::free_balance(&BOB);
+                assert_eq!(bobs_balance, expected_bobs_balance,
+                    "BOB's balance should be updated after selfdestruct for {:?}", fixture_type);
+            }
+            assert!(test_utils::get_contract_checked(&addr).is_none());
+		});
+    }
+
+    // TODO: how to call selfdestruct from solidity in PVM?
```

Comment:
let's fix this and do what we said yesterday that we will just keep the selfdestruct behavior of EVM for now